### PR TITLE
update generic link text guidance

### PR DESCRIPTION
**Pull request by khiga8**
Closes: https://github.com/github/accessibility/issues/1343 (staff-only link)

This PR updates the link guidance with more guidance around generic link text and techniques. 

Updating this to make sure we document the discussions we've been having on Slack and this [recommended guidance in a recent GitHub issue](https://github.com/github/accessibility/issues/1291#issuecomment-1159088348) (staff-only link).

### CODE REVIEW

**Review by yaili — CHANGES_REQUESTED** [review]

File: content/accessibility/links.mdx [hunk 1]
```diff
@@ -67,10 +67,51 @@ People who have low or colorblind vision may have trouble identifying links that
 - Only adding an underline to a link when it has focus or is hovered over
 - Underlining every link on the page. For example, a navigation list is a list of links but navigational links don't have to be underlined because the intent is understood and an underline is not needed to identify the interactive nature of the control.
 
+## Link text
```
Comment (owenniblock) [inline content/accessibility/links.mdx:70]: Suggest a slightly more descriptive title here:

```suggestion
## Writing link text
```

File: content/accessibility/links.mdx [hunk 2]
```diff
@@ -16,7 +16,7 @@ People who have low or colorblind vision may have trouble identifying links that
 
 ## How to test links
 - Check a link's function. If clicking a link activates a feature on the page, it should be a `<button>`, and vice versa. 
-- Check a link's purpose. If you can't get an idea of where a link will take you based on the link text alone, the link text should be updated. 
+- Check a link's purpose. If you can't get an idea of where a link will take you based on the link text alone, the link text should be updated. Learn more at [Link text](#link-text).
```
Comment (owenniblock) [inline content/accessibility/links.mdx:19]: Related to my suggestion to change the title below.

```suggestion
- Check a link's purpose. If you can't get an idea of where a link will take you based on the link text alone, the link text should be updated. Learn more at [Writing link text](#writing-link-text).
```

**Review by owenniblock — APPROVED** [review]
Minor niggle otherwise looks good 👍

File: content/accessibility/links.mdx [hunk 3]
```diff
@@ -16,7 +16,7 @@ People who have low or colorblind vision may have trouble identifying links that
 
 ## How to test links
 - Check a link's function. If clicking a link activates a feature on the page, it should be a `<button>`, and vice versa. 
-- Check a link's purpose. If you can't get an idea of where a link will take you based on the link text alone, the link text should be updated. 
+- Check a link's purpose. If you can't get an idea of where a link will take you based on the link text alone, the link text should be updated. Learn more about [link text](#link-text).
```
[the remark this answers is not in the document]
Comment (ichelsea) [reply content/accessibility/links.mdx:19]:
```suggestion
- Check a link's purpose. If you can't get an idea of where a link will take you based on the link text alone, the link text should be updated. [Learn how to write good link text](#writing-link-text).
```

File: content/accessibility/links.mdx [hunk 4]
```diff
@@ -67,10 +67,51 @@ People who have low or colorblind vision may have trouble identifying links that
...
+### Good: Accessible name includes visible link text 👍 
+
+The link below has an accessible name that fully includes the visible link text and is comprehensible.
+(This example uses the ARIA technique for demo purposes and is technically valid, but the link text here can just be "Learn more about GitHub pricing plans".)
+
+```HTML
+There are various plans available. <a href="..." aria-label="Learn more about GitHub pricing plans">Learn more</button>
+```
+
+As demonstrated in the examples, this technique adds more complexity to the code and can introduce more problems than it solves so only use this technique if absolutely necessary.
```
[the remark this answers is not in the document]
Comment (khiga8) [reply content/accessibility/links.mdx:108]: ooh I only saw examples that use screenshots. thanks for the link using a code example.

[the remark this answers is not in the document]
Comment (khiga8) [reply content/accessibility/links.mdx]: maybe we should introduce linters against, "bad", "good", and emojis.

**Comment by khiga8** [comment]
thank you for your feedback @yaili ! mind another review?
I converted the examples to use the DoDont container and got rid of the emojis/banned words.

**Review by yaili — APPROVED** [review]
Thank you! 😊

File: content/accessibility/links.mdx [hunk 5]
```diff
@@ -69,30 +69,43 @@ People who have low or colorblind vision may have trouble identifying links that
...
 It may be acceptable in some scenarios to provide a more descriptive link text for screen reader users by setting the `aria-label`. However, this technique will result in divergence between the label and the text and is not an ideal, future-proof solution.
 Whenever possible, prefer a single descriptive link text that is available for both sighted users and screen reader users.
 
-If you must resort to the ARIA technique to provide a descriptive link text, you must follow these principles:
+If you must resort to the ARIA technique to provide a descriptive link text, follow these principles:
 
 1. The visible text is included in full as part of the accessible name to avoid a violation of [SC 2.5.3: Label in Name](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html).
 2. The sentence that the link is a part of is well-formed and grammatical when read with both the visible text and the accessible name.
 
-### Bad: Accessible name results in badly formed sentence 👎
```
Comment (Duytran889889) [inline content/accessibility/links.mdx]: Hsndn d đn
[Uploading guide (1).pdf…]()

**Review by Duytran889889 — COMMENTED** [review]
Hhdjjd

**Pull request merged**